controller/internal/enforcer/proxy/enforcerproxy.go

```diff
@@ -68,6 +68,10 @@ func (s *ProxyInfo) InitRemoteEnforcer(contextID string) error {
 		return fmt.Errorf("failed to initialize remote enforcer: status: %s: %s", resp.Status, err)
 	}
 
+	if resp.Status != "" {
+		zap.L().Warn("received status while initializing the remote enforcer", zap.String("contextID", resp.Status))
+	}
+
 	s.Lock()
```

Comment:
What is the scenario where the status is interesting if there is no error? I actually think that we should only check the status if there is an error and insert it in the error message so that callers will know why it failed.

Reply:
While reading the RPC code, I got confused because we return an error but also a status, which is always redundant, and usually the same. Why are those separate, ans why do we need a status?

From what I can see the only time we use a status is also when we return an error.

Reply:
The status will not be populated if there is an error. The response object is not sent back to the client if an error is returned from this method

Reply:
that's not what I'm seeing. Both the error and status are successfully returned. Hence my question of why do we need those ?

Reply:
net/rpc doc says if you return non-nil error response object will not be populated. 
The idea was to always return nil from remote enforcer and only use the resp.Status to return errors from remote enforcer and use the err to indicate rpc channel failure or encoder issues